test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1883,6 +1884,37 @@
            (testing "concurrent-depth metric should be 0"
              (is (= 0 (concurrent-depth))))))))))
 
+(deftest block-command-processing
+  (with-test-db
+    (svc-utils/with-pdb-with-no-gc
+      (let [dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+            stats (-> dispatcher service-context :stats)
+            new-producer-ts (now)
+            base-cmd (get-in wire-catalogs [9 :basic])
+            enqueue-catalog (fn []
+                              (enqueue-command
+                                dispatcher
+                                (command-names :replace-catalog)
+                                9
+                                (:certname base-cmd)
+                                nil
+                                (->  base-cmd
+                                     (assoc :producer_timestamp new-producer-ts)
+                                     tqueue/coerce-to-stream)
+                                ""))]
+
+        (testing "should pause execution"
+          (pause-execution dispatcher)
+          (enqueue-catalog)
+
+          (is (= :timeout (await-ref-state stats #(= (% :executed-commands) 1), 500 :timeout))))
+
+        (testing "should resume execution"
+          (resume-execution dispatcher)
+
+          (is (not= :timeout (await-ref-state stats #(= (% :executed-commands) 1), default-timeout-ms :timeout))))))))
```

Comment:
Suggest wrapping this somehow, at say 80-100 chars.  Also suggest removing commas (they're generally omitted).

Reply:
Removed commas and wrapped thee line.